pkgs/development/python-modules/deal/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+
+  # build
+, flit-core
+, astroid
+
+  # tests
+, pytestCheckHook
+, docstring-parser
+, isort
+, marshmallow
+, pytest-cov
+, sphinx
+, hypothesis
+, vaa
+, deal-solver
+, pygments
+, typeguard
+, coverage
+, urllib3
+}:
+
+buildPythonPackage rec {
+  pname = "deal";
+  version = "4.23.3";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "life4";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    hash = "sha256-duFxe2KSQQb7HB5KrrE32xzTb6QkQcrQssiuXLKao50=";
+  };
+
+  preCheck = ''
+    # don't do coverage
+    substituteInPlace pyproject.toml \
+      --replace "\"--cov-fail-under=100\"," "" \
+      --replace "\"--cov=deal\"," "" \
+      --replace "\"--cov-report=html\"," "" \
+      --replace "\"--cov-report=term-missing:skip-covered\"," ""
+  '';
+
+  nativeBuildInputs = [
+    flit-core
+  ];
+
+  propagatedBuildInputs = [
+    astroid
+    deal-solver
+    pygments
+    typeguard
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+
+    docstring-parser
+    isort
+    marshmallow
+    pytest-cov
+    sphinx
+    hypothesis
+    vaa
+    coverage
+    urllib3
+  ];
+
+  disabledTests = [
+    # needs internet access
+    "test_smoke_has"
+    "test_pure_offline"
+    "test_raises_doesnt_override_another_contract"
+    "test_raises_doesnt_override_another_contract_async"
+    "test_raises_generator"
+    # AttributeError: module 'vaa' has no attribute 'Error'
```

Comment:
Does this impact runtime? This could mean things are broken.

Reply:
Good question. Unfortunately, deal doesn't use any kind of issue tracker (AFAIK). The only PR I found regarding `vaa` is https://github.com/life4/deal/pull/117 which makes its usage optional. I guess it won't impact the usage of deal, but I can't be sure.

Reply:
Okay, hopefully this works.